### PR TITLE
Suppress `MatrixTest` on Windows CI

### PR DESCRIPTION
**Pull request by jglick**
Does not solve https://github.com/jenkinsci/bom/issues/613 but may at least restore CI in this repo: https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/477#issuecomment-1007614950

### CODE REVIEW

File: pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/MatrixTest.java [hunk 1]
```diff
@@ -66,6 +68,7 @@
     private static String password;
     @BeforeClass
     public static void setUpAgent() throws Exception {
+        assumeFalse("TODO frequent failures in CI on Windows", Functions.isWindows());
```
Comment (basil) [inline pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/MatrixTest.java:71]:
```suggestion
        assumeFalse("TODO frequent failures in CI on Windows", Functions.isWindows() && System.getenv("CI") != null);
```

**Review by basil — APPROVED** [review]
:man_shrugging: If it passes

**Comment by basil** [comment]
> If it passes

But it doesn't, while #477 does. :man_shrugging:

**Comment by jglick** [comment]
Hmm, that is a different error than other builds have though.

**Comment by basil** [comment]
So if we lack the time/motivation to take this PR to the finish line for the time being, why not take #477 and deal with this later when it is practical?

**Comment by jglick** [comment]
Probably going to leave that for @car-roll.

**Comment by jtnord** [comment]
> C:\Jenkins\workspace\e-model-definition-plugin_master\pipeline-model-definition\target\tmp\j h3483184721116739069\workspace\test12\subdir>mvn --version 
Error occurred during initialization of VM
Could not reserve enough space for object heap


and we are using [4GB heap for surefire the tests](https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/2f4089bc15fb391769eb4dd439b573372301df98/pom.xml#L167)?!?

**Comment by jtnord** [comment]
see also #479 and jenkinsci/jenkins-test-harness#380

**Pull request closed without merging**